templates/CRM/common/contactFooter.tpl

```diff
@@ -13,6 +13,7 @@
   <span class="col1">
     {if $external_identifier}{ts}External ID{/ts}:&nbsp;{$external_identifier}{/if}
     {if $action !== 2}&nbsp; &nbsp;{ts}Contact ID{/ts}:&nbsp;{$contactId}{/if}
+    {if $userRecordUrl}&nbsp; &nbsp;{ts}User ID{/ts}:&nbsp;<a title="{ts}View user record{/ts}" class="user-record-link" href="{$userRecordUrl}">{$userRecordId}</a>{/if}
```

Comment:
The reason tests are failing is `$userRecordUrl` in general we go to where the value is assigned & ensure it is always assigned - in this case the template is used so broadly it proably has to go in this array https://github.com/civicrm/civicrm-core/blob/85aa1eb6213781fac25ff288cd034b4dcde5e6ae/CRM/Core/Form.php#L256-L293

Reply:
@andyburnsco it sounds like there is general agreement with this PR and it just needs this one change to get tests passing.